### PR TITLE
[WIP] Implementing the Data Acquisition files using python

### LINKED ISSUE #2: Use sbt-jmh for benchmarking
Better benchmarking of the library.

Use ```sbt-jmh``` to benchmark and visualise the performance.

This was a recommendation from a [plokhotnyuk](https://www.reddit.com/user/plokhotnyuk)

### PR DESCRIPTION
**Pull request by dandxy89**
TODOs:

- [X] Create Python scripts to generate data
- [X] Update existing benchmark scripts
- [X] Identify suitable benchmarking tool for Python
- [ ] Run benchmarks 
- [ ] Review by @inejc - resolves #2

Optional:

- [ ] Include the flame graph functionality - will move to another PR / issue

### CODE REVIEW

File: src/main/scala/com/picnicml/doddlemodel/linear/JMH_LinearRegression.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.picnicml.doddlemodel.linear
+
+import breeze.linalg.DenseMatrix
+import com.picnicml.doddlemodel.data.loadCsvDataset
+import org.openjdk.jmh.annotations.Benchmark
+
+class JMH_LinearRegression {
+
+  import LinearFileData._
+
```
Comment (dandxy89) [inline src/main/scala/com/picnicml/doddlemodel/linear/JMH_LinearRegression.scala]: @inejc nice syntax for benchmarking

File: README.md [hunk 2]
```diff
@@ -1,94 +1,66 @@
...
+
+The baseline dataset that are already may not be substantially large enough to for benchmarking therefor a second script **RunGenerator.py** to make either classification or regression dataset. Edit the parameters in the script and then run. The parameters that are defined are the default used to benchmark the two implementations.
+
+# Linear Regression
+
+Dataset Name:   **Regression_data.csv**
+Sample Size:    100,000
+
+Run benchmarks - ```jmh:run -bm AverageTime -i 20 -wi 20 -f1 -t1 .*JMH_Linear*```
+
```
Comment (dandxy89) [inline README.md:43]: @inejc do you know of a good benchmarking tool in Python. Happy to use the time.time() ...

Comment (inejc) [reply README.md]: @dandxy89 I haven't used anything other than `time`. Perhaps we could write a function decorator to make functionality reusable?

Comment (inejc) [reply README.md:43]: @dandxy89 I haven't used anything other than time. Perhaps we could write a function decorator to make functionality reusable?

File: build.sbt [hunk 3]
```diff
@@ -6,3 +6,5 @@ lazy val root = (project in file("."))
     scalaVersion := "2.12.6",
     libraryDependencies ++= Dependencies.settings
```
Comment (inejc) [inline build.sbt]: Is there another dependency that we have to add to `Dependencies.scala` or is this just a plugin?

File: src/main/scala/com/picnicml/doddlemodel/linear/JMH_LinearRegression.scala [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package com.picnicml.doddlemodel.linear
+
+import breeze.linalg.DenseMatrix
+import com.picnicml.doddlemodel.data.loadCsvDataset
+import org.openjdk.jmh.annotations.Benchmark
+
+class JMH_LinearRegression {
```
Comment (inejc) [inline src/main/scala/com/picnicml/doddlemodel/linear/JMH_LinearRegression.scala]: What is the purpose of `_` in `JMH_LinearRegression`? Should we just name it `JMHLinearRegression` to be consistent with other style choices, or is there a reason for that?

**Review by inejc — COMMENTED** [review]
Thanks for the PR :). So far so good. I haven't tried running the benchmarks yet. Will do so when all of the functionality is implemented.

File: README.md (hunk 2 above)
Comment (inejc) [reply README.md:43]: A quick search yielded this: https://github.com/python/performance. Not sure if we need it though? We could write a simple decorator that runs a function `n` times and returns mean and avg?

File: build.sbt (hunk 3 above)
Comment (dandxy89) [reply build.sbt]: Just a plug-in

File: src/main/scala/com/picnicml/doddlemodel/linear/JMH_LinearRegression.scala (hunk 4 above)
Comment (dandxy89) [reply src/main/scala/com/picnicml/doddlemodel/linear/JMH_LinearRegression.scala]: To be honest, no reason other than it was inherited from the examples. 😉

Comment (inejc) [reply src/main/scala/com/picnicml/doddlemodel/linear/JMH_LinearRegression.scala]: Ok, I guess we should use `JMHLinearRegression` then :).

**Comment by inejc** [comment]
Another thing that would be useful to benchmark in the future is cross-validation and hyperparameter search. I'm really interested in how it compares to scikit.

**Pull request closed without merging**